File: forum-backend/src/main/java/com/projects/pes/forumbackend/pojo/Resource.java
package com.projects.pes.forumbackend.pojo;

import com.projects.pes.forumbackend.entities.ForumEntity;
import com.projects.pes.forumbackend.entities.PostEntity;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToMany;

import java.util.Date;
import java.util.Set;
import java.util.UUID;

public record Resource(
         UUID id,
         String name,
         String ownerName,
         UUID validatorId,
         Boolean validated,
         Date dateOfPublish,
         String contentType,
         byte[] content,
         String url
) {
}

File: student-forums-frontend/src/app/resource.ts
export class Resource {
    id:string;
    name:string;
    ownerName:string;
    validatorId:string;
    validated:boolean;
    dateOfPublish:Date;
    contentType:string;
    url:string;
    constructor(id:string, name:string, ownerName:string, validatorId:string, validated:boolean, dateOfPublish:Date,contentType:string,url:string) {
        this.id = id;
        this.name = name;
        this.ownerName = ownerName;
        this.validated = validated;
        this.validatorId = validatorId;
        this.dateOfPublish = dateOfPublish;
        this.contentType = contentType;
        this.url = url;

    }
}
